indico/modules/rb_new/client/js/common/bookings/BookingDetailsModal.jsx

```diff
@@ -160,33 +171,186 @@ class BookingDetailsModal extends React.Component {
-            )}
-            {showAccept && (
-                <Button type="button"
-                        icon="check circle"
-                        color="green"
-                        size="small"
-                        content={<Translate>Accept booking</Translate>} />
-            )}
-        </Modal.Actions>
-    );
+    renderActionButtons = (canCancel, canReject, showAccept) => {
+        const {bookingStateChangeInProgress} = this.props;
+        const {actionInProgress, openConfirm} = this.state;
+        const rejectButton = (
+            <Button type="button"
+                    icon="remove circle"
+                    color="red"
+                    size="small"
+                    loading={actionInProgress === 'reject' && bookingStateChangeInProgress}
+                    disabled={bookingStateChangeInProgress}
+                    content={<Translate>Reject booking</Translate>} />
+        );
+
+        const renderForm = ({handleSubmit, hasValidationErrors, submitSucceeded, submitting, pristine}) => {
+            return (
+                <Form styleName="rejection-form" onSubmit={handleSubmit}>
+                    <Field name="reason"
+                           component={ReduxFormField}
+                           as={TextArea}
+                           format={formatters.trim}
+                           placeholder={Translate.string('Provide the rejection reason')}
+                           rows={2}
+                           validate={validators.required}
+                           disabled={submitting}
+                           required
+                           formatOnBlur />
+                    <Button type="submit"
+                            disabled={submitting || pristine || hasValidationErrors || submitSucceeded}
+                            loading={submitting}
+                            floated="right"
+                            primary>
+                        <Translate>Reject</Translate>
+                    </Button>
+                </Form>
+            );
+        };
+
+        return (
+            <Modal.Actions>
+                {canCancel && (
+                    <>
+                        <Button type="button"
+                                icon="cancel"
+                                size="small"
+                                onClick={() => this.showConfirm('cancel')}
+                                loading={actionInProgress === 'cancel' && bookingStateChangeInProgress}
+                                disabled={bookingStateChangeInProgress}
+                                content={<Translate>Cancel booking</Translate>} />
+                        <Confirm header={Translate.string('Confirm cancellation')}
+                                 content={Translate.string('Are you sure you want to cancel this booking?' +
+                                                           ' This will cancel all occurrences of this booking.')}
```

Comment:
what about keeping the trailing space in the previous line like we do in python?